src/ncp/ncp_base.hpp

```diff
@@ -101,13 +106,75 @@ class NcpBase
     /**
      * This static method returns the pointer to the single NCP instance.
      *
-     @returns Pointer to the single NCP instance.
+     * @returns Pointer to the single NCP instance.
      *
      */
     static NcpBase *GetNcpInstance(void);
 
-protected:
+    /**
+     * This method sends data to host via specifiec stream.
+     *
+     *
+     * @param[in]  aStreamId  A numeric identifier for the stream to write to.
+     *                        If set to '0', will default to the debug stream.
+     * @param[in]  aDataPtr   A pointer to the data to send on the stream.
+     *                        If aDataLen is non-zero, this param MUST NOT be NULL.
+     * @param[in]  aDataLen   The number of bytes of data from aDataPtr to send.
+     *
+     * @retval OT_ERROR_NONE         The data was queued for delivery to the host.
+     * @retval OT_ERROR_BUSY         There are not enough resources to complete this
+     *                               request. This is usually a temporary condition.
+     * @retval OT_ERROR_INVALID_ARGS The given aStreamId was invalid.
+     *
+     */
+    otError StreamWrite(int aStreamId, const uint8_t *aDataPtr, int aDataLen);
+
+#if OPENTHREAD_MTD || OPENTHREAD_FTD
+#if OPENTHREAD_CONFIG_NCP_ENABLE_PEEK_POKE
```

Comment:
Peek/poke support can be useful in general. How about not limiting them to `MTD` and `FTD`...?